rfc/rfc-43/rfc-43.md

```diff
@@ -0,0 +1,316 @@
+      | schedule_time  | timestamp | operation created time                             |
+      | create_time    | timestamp | operation created time                             |
+      | update_time    | timestamp | operation updated time                             |
+
+## Interface design
+
+### Register
+
+/v1/hoodie/service/register
+
+```
+{
+"table_name":"table_name",
+"base_path":"/hoodie/base_path",
+"owner":"default",
+"queue":"root.default"
+"priority":"0"
+}
+```
+
+## Error Handling
+
+Due to two tables being involved in each operation, we need to ensure that they are in sync and errors during dataset
+operation are handled correctly.
+
+1. Client scheduled the plan but request failed.
+
+2. RequestHandler received request but the commit is not completed.
+
+3. Client rollback plan after request to Compaction/Clustering Service.
+
+Client will report to service all unCompleted plans by time or num deltaCommits or combination, the service upsert to
+meta table uses HoodieServicePayload to guarantee any plan unique. Client use Delete request to rollback plan for
+Compaction/Clustering Service.
+
+### Configuration
+
+hoodie.table.services.enable=true
+hoodie.table.service.compaction.enable=true
+hoodie.table.service.clustering.enable=true
+hoodie.table.service.clean.enable=true
+
+## Proposal
+
+### phase1
+
+Implement the basic functional part of the long-term planning without enabling meta server
+
+1. Processing mode + processing flow
+2. Storage
+3. Execution (only implements Spark )
+4. Monitoring (only expose basic success, failure and other indicators)
+5. API（only REST）
+6. Writer
+
+**Landing plan: 0.12**
+
+### Phase2
+
+Realize the integration part with the meta server and improve some capabilities
+
+1. Improve monitoring and alarm indicators and provide alarm functions
+2. API （ GRPC ）
+3. Schedule table service in Table Management Service（with meta server）
+4. Multi-instance implementation
+
+**Landing plan: 1.0**
+
+## Rollout/Adoption Plan
+
+- What impact (if any) will there be on existing users?
+    - None. This service will start in another process. If the config is not set, the functionality will not be enabled.
+
+- If we are changing behavior how will we phase out the older behavior?
+    - This new feature is backwards-compatible. If the compaction/clustering service are not available, the existing way
+      like inline/async will be used.
+
+- If we need special migration tools, describe them here.
+    - NA
+
+- When will we remove the existing behavior?
+    - It is not required.
+
+Test Plan
+
+This RFC aims to implement a new Service to manager the Hudi table compaction and clustering action, to test this
+feature, there will be some test tables trigger compaction and clustering action to Service with unit tests for the
+code. Since this is an entirely new feature, I am confident that this will not cause any regressions during and after
+roll out.
```

Comment:
Test plan is not only about regression or impacting existing services. For this feature itself, what kinds of tests can be carried out? what scenarios will be covered in the OSS CI? please re-work on this section.